csharp/src/Google.Protobuf/CodedInputStream.cs

```diff
@@ -470,87 +521,259 @@ internal void SkipGroup(uint startGroupTag)
         /// <summary>
         /// Reads a double field from the stream.
         /// </summary>
+        [SecuritySafeCritical]
         public double ReadDouble()
         {
-            return BitConverter.Int64BitsToDouble((long) ReadRawLittleEndian64());
+            var immediateBudder = ImmediateBuffer;
+            return ReadDouble(ref immediateBudder);
         }
 
         /// <summary>
-        /// Reads a float field from the stream.
+        /// This supports the Protocol Buffers infrastructure and is not meant to be used directly from your code.
         /// </summary>
-        public float ReadFloat()
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [SecurityCritical]
+        public double ReadDouble(ref ReadOnlySpan<byte> immediateBuffer) => BitConverter.Int64BitsToDouble((long)ReadRawLittleEndian64(ref immediateBuffer));
+
+        /// <summary>
+        /// This supports the Protocol Buffers infrastructure and is not meant to be used directly from your code.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [SecurityCritical]
+        public double ReadWrappedDouble(ref ReadOnlySpan<byte> immediateBuffer)

```

Comment:
is this needed for the change to remove codecs? If so, we should leave this change for a followup PR.

Reply:
Hmmm we can split removing codecs and not using codecs as two PRs. Wrapped versions are needed for the new codegen